### PR TITLE
set up nightwatch

### PR DESCRIPTION
**Pull request by yneves**
Here's how it's working:
 * e2e tests are located in `tests/*.spec.js`
 * using http://nightwatchjs.org/ with selenium and chromedriver
 * `npm test` downloads binaries and run nightwatch test runner
 * need to add `nightwatch.globals.json` file with email and password to be used in the tests

Next steps:
 * Run with phantomjs - so we can run on CI
 * Run with firefox, safari, edge and ie - not so important, but nice to have

### CODE REVIEW

**Comment by yneves** [comment]
@ntgn81 Let me know what you think. I removed the karma stuff there since it doesn't seem to be used. I'll add unit tests with [enzyme](http://airbnb.io/enzyme/) + mocha later.

**Comment by ntgn81** [comment]
@yneves , this looks pretty nice. A few points though:

- chromedriver_mac32.zip wouldn't work anywhere but mac - it didn't work for me on windows. And it probably won't work on CI environment since it's not Mac. Can we use something like phantomjs instead?
- Karma is used to run (our non-existent) unit tests, I think we should keep using that.
- And I think it might be better to target unit tests first, what do you think?

**Comment by yneves** [comment]
@ntgn81 Can you give it another try? Just run `npm test` and it will:

 * compile
 * download binaries
 * start http server at port 4000
 * run nightwatch
 * stop http server

**Comment by yneves** [comment]
I mean, all with phantomjs now.

File: src/index.html [hunk 1]
```diff
@@ -7,5 +7,25 @@
 </head>
 <body>
   <div id="root" class="root"></div>
+  <script>
```
Comment (yneves) [inline src/index.html]: Had to add this here because i couldn't find a better way of loading it just for phantomjs. Seems pretty much harmless anyway.

Comment (ntgn81) [inline src/index.html]: phantomjs doesn't have .bind? That's pretty weird lol

Comment (yneves) [inline src/index.html]: Yeah, crazy. I copied this from `phantomjs-polyfill` module in our dependencies, it was already there.

**Comment by yneves** [comment]
`npm test` now is just running lint until i get the unit tests running, `npm run test:e2e` will run the nightwatch tests

**Pull request merged**